### DISCUSSION TITLE
Not able to find machines with AND operator

### DISCUSSION THREAD

**Discussion by sunilkumar-nfer** [prometheus/prometheus#12796] [Q&A]
### What did you do?

we have 2k VMs where node exporter is running , We want to know all the VMs where these conditions are f fulfilled 

- CPU idle time is greater than 90%mem
- Free memory is above 85%
- Low disk I/O activity on 'sda'
- Low network traffic (receive and transmit below 1,000,000 bytes)

but using the AND operator we are not able to find those.

taken as example 

100 - (100 * (node_memory_MemAvailable_bytes / node_memory_MemTotal_bytes)) > 70 and avg_over_time(node_cpu_seconds_total{mode="idle"}[5m]) < 30

this query should return the VMs with CPU usage greater than 70% and memory usage greater than 70 percent

there are many machines that fulfil that conditions but we are not able to find those machines via promQL

Can anyone help here.

### What did you expect to see?

Above query should return the VMs that fulfil that conditions .

### What did you see instead? Under which circumstances?

the above query give us empty result

**Comment by roidelapluie** [comment]
This is what ChatGPT gave me:

```
(
  100 - avg_over_time(node_cpu_seconds_total{mode="idle"}[5m]) * 100 < 10
) and (
  (node_memory_MemAvailable_bytes / node_memory_MemTotal_bytes) * 100 > 85
) and (
  rate(node_disk_io_time_seconds_total{device="sda"}[5m]) < 0.1
) and (
  (rate(node_network_receive_bytes_total[5m]) + rate(node_network_transmit_bytes_total[5m])) < 1000000
)
```

(I have not checked the answer)

**Comment by sunilkumar-nfer** [comment]
@roidelapluie, i try this, but it also gives me an empty query result. can you try and check whether this query working fine with only conditions of  CPU and RAM?
